api/receiver/receive_results.go

```diff
@@ -62,33 +62,38 @@ func HandleResultsUpload(a API, w http.ResponseWriter, r *http.Request) {
 
 	log := shared.GetLogger(a.Context())
 	var results, screenshots []string
-	if r.MultipartForm != nil && r.MultipartForm.File != nil && len(r.MultipartForm.File["result_file"]) > 0 {
+	if f := r.MultipartForm; f != nil && f.File != nil && len(f.File["result_file"]) > 0 {
 		// result_file[] payload
-		files := r.MultipartForm.File["result_file"]
-		sFiles := r.MultipartForm.File["screenshot_file"]
+		files := f.File["result_file"]
+		sFiles := f.File["screenshot_file"]
 		log.Debugf("Found %d result files, %d screenshot files", results, screenshots)
 		var err error
 		results, screenshots, err = saveToGCS(a, uploader, files, sFiles)
 		if err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
+			log.Errorf("Failed to save files to GCS: %v", err)
+			http.Error(w, "Failed to save files to GCS", http.StatusInternalServerError)
 			return
 		}
 	} else if artifactName := getAzureArtifactName(r.PostForm.Get("result_url")); artifactName != "" {
 		// Special Azure case for result_url payload
 		azureURL := r.PostForm.Get("result_url")
 		log.Debugf("Found Azure URL: %s", azureURL)
 		extraParams["azure_url"] = azureURL
-	} else {
+	} else if len(r.PostForm["result_url"]) > 0 {
 		// General result_url[] payload
 		results = r.PostForm["result_url"]
 		screenshots = r.PostForm["screenshot_url"]
 		log.Debugf("Found %d result URLs, %d screenshot URLs", results, screenshots)
+	} else {
+		log.Errorf("No results found")
```

Comment:
Will it be clear on the logs side what this refers to? (e.g. if it shows up as `receive_results.go: No results found` thats probably clear, but if it just shows up as `[timestamp] No results found` thats quite hard to debug from!

Reply:
Yeah we should probably improve the underlying log handler to include file names, but that'd be another PR.